miasm2/arch/aarch64/arch.py

```
@@ -0,0 +1,1968 @@
+            return '[%s, %s]' % (e.args[0], arg)
+
+        fds
+        return o
+
+    def dstflow(self):
+        return self.name in self.name in brcond + ["B", "BL"]
+
+    def mnemo_flow_to_dst_index(self, name):
+        if self.name in ['CBZ', 'CBNZ']:
+            return 1
+        elif self.name in ['TBZ', 'TBNZ']:
+            return 2
+        else:
+            return 0
+
+    def dstflow2label(self, symbol_pool):
+        index = self.mnemo_flow_to_dst_index(self.name)
+        e = self.args[index]
+        if not isinstance(e, m2_expr.ExprInt):
+            return
+        ad = e.arg + self.offset
+        l = symbol_pool.getby_offset_create(ad)
+        s = m2_expr.ExprId(l, e.size)
+        self.args[index] = s
+
+    def breakflow(self):
+        if self.name in brcond + ["BR", "BLR", "RET", "ERET", "DRPS", "B", "BL"]:
+            return True
+        return False
+
+    def is_subcall(self):
+        if self.name in ["BLR", "BL"]:
+            return True
+        return False
+
+    def getdstflow(self, symbol_pool):
+        index = self.mnemo_flow_to_dst_index(self.name)
+        return [self.args[index]]
+
+    def splitflow(self):
+        if self.name in brcond + ["BLR", "BL"]:
+            return True
+        return False
+
+    def get_symbol_size(self, symbol, symbol_pool):
+        return 64
+
+    def fixDstOffset(self):
+        index = self.mnemo_flow_to_dst_index(self.name)
+        e = self.args[index]
+        if self.offset is None:
+            raise ValueError('symbol not resolved %s' % l)
+        if not isinstance(e, m2_expr.ExprInt):
+            log.debug('dyn dst %r', e)
+            return
+        off = e.arg - self.offset
+        if int(off % 4):
+            raise ValueError('strange offset! %r' % off)
+        self.args[index] = m2_expr.ExprInt32(off)
+
+    def get_args_expr(self):
+        fds
+        args = [a for a in self.args]
+        return args
+
+    def get_asm_offset(self, expr):
+        fds
```

You should raise an error instead